koku/masu/database/trino_sql/gcp/openshift/managed_gcp_openshift_daily.sql

```diff
@@ -0,0 +1,159 @@
+    usage_amount_in_pricing_units double,
+    currency varchar,
+    cost double,
+    daily_credits double,
+    resource_global_name varchar,
+    resource_id_matched boolean,
+    matched_tag varchar,
+    source varchar,
+    ocp_source varchar,
+    year varchar,
+    month varchar,
+    day varchar
+) WITH(format = 'PARQUET', partitioned_by=ARRAY['source', 'ocp_source', 'year', 'month', 'day'])
+;
+
+-- Direct resource matching
+INSERT INTO hive.{{schema | sqlsafe}}.managed_gcp_openshift_daily (
+    invoice_month,
+    billing_account_id,
+    project_id,
+    usage_start_time,
+    service_id,
+    sku_id,
+    system_labels,
+    labels,
+    cost_type,
+    location_region,
+    resource_name,
+    project_name,
+    service_description,
+    sku_description,
+    usage_pricing_unit,
+    usage_amount_in_pricing_units,
+    currency,
+    cost,
+    daily_credits,
+    resource_global_name,
+    resource_id_matched,
+    matched_tag,
+    source,
+    ocp_source,
+    year,
+    month,
+    day
+)
+WITH cte_gcp_resource_names AS (
+    SELECT DISTINCT resource_name, service_description
+    FROM hive.{{schema | sqlsafe}}.gcp_line_items_daily
+    WHERE source = {{gcp_source_uuid}}
+        AND year = {{year}}
+        AND month = {{month}}
+        AND usage_start_time >= {{start_date}}
+        AND usage_start_time < date_add('day', 1, {{end_date}})
+),
+cte_array_agg_nodes AS (
+    SELECT DISTINCT node
+    FROM hive.{{schema | sqlsafe}}.openshift_pod_usage_line_items_daily
+    WHERE source = {{ocp_source_uuid}}
+        AND year = {{year}}
+        AND month = {{month}}
+        AND interval_start >= {{start_date}}
+        AND interval_start < date_add('day', 1, {{end_date}})
+),
+cte_array_agg_volumes AS (
+    SELECT DISTINCT persistentvolume, csi_volume_handle
+    FROM hive.{{schema | sqlsafe}}.openshift_storage_usage_line_items_daily
+    WHERE source = {{ocp_source_uuid}}
+        AND year = {{year}}
+        AND month = {{month}}
+        AND interval_start >= {{start_date}}
+        AND interval_start < date_add('day', 1, {{end_date}})
+),
+cte_matchable_resource_names AS (
+    SELECT resource_names.resource_name, resource_names.service_description
+    FROM cte_gcp_resource_names AS resource_names
+    JOIN cte_array_agg_nodes AS nodes
+        ON strpos(resource_names.resource_name, nodes.node) != 0
+
+    UNION
+
+    SELECT resource_names.resource_name, resource_names.service_description
+    FROM cte_gcp_resource_names AS resource_names
+    JOIN cte_array_agg_volumes AS volumes
+        ON (
+            strpos(resource_names.resource_name, volumes.persistentvolume) != 0
+            OR strpos(resource_names.resource_name, volumes.csi_volume_handle) != 0
+        )
+
+),
+cte_tag_matches AS (
+  SELECT * FROM unnest(ARRAY{{matched_tag_array | sqlsafe}}) as t(matched_tag)
+
+  UNION
+
+  SELECT * FROM unnest(ARRAY['openshift_cluster', 'openshift_node', 'openshift_project']) as t(matched_tag)
```

Comment:
I am not sure if we are following the logic we are currently doing in python here. I wonder if we are leaving out this [section](https://github.com/project-koku/koku/blob/main/koku/masu/util/gcp/common.py#L97). This section appears to be what we are calling the [special_case_tag_matched](https://github.com/project-koku/koku/blob/main/koku/masu/util/gcp/common.py#L109-L117).

Reply:
Skipping this for now. :smile: